### PR TITLE
Handle the new Low/Medium/High/Critical in pysemgrep

### PR DESCRIPTION
**Pull request by aryx**
test plan:
e2e test included
cd cli; make e2e

### CODE REVIEW

**Comment by aryx** [comment]
followup to #10165

**Comment by aryx** [comment]
I didn't add a changelog entry on purpose; This is not yet official; we just want to add support for it ASAP
so when we change rules to use critical, hopefully enough old versions will be able to handle it already.

**Review by p4p3r — APPROVED** [review]
lgtm

File: cli/tests/default/e2e/test_output.py [hunk 1]
```diff
@@ -308,42 +308,6 @@ def test_omit_experiment(run_semgrep_in_tmp: RunSemgrep, snapshot):
     snapshot.assert_match(stdout, "results.out")
 
 
-@pytest.mark.kinda_slow
```
Comment (aryx) [inline cli/tests/default/e2e/test_output.py:311]: @mjambon I've removed this test_debug_experimental_rule test.
It's not useful and worth it's annoying to udpate indeed as you wrote below.

**Pull request merged**